docs/architecture-decisions/adr011-plugin-package-structure.md

```diff
@@ -0,0 +1,74 @@
+---
+id: adrs-adr011
+title: ADR011: Plugin Package Structure
+description: Architecture Decision Record (ADR) for Plugin Package Structure
+---
+
+## Context
+
+A core feature of Backstage is the extensibility via plugins. The Backstage
+repository is open for contributions of plugins. Even most of the core features
+are implemented as plugins. A plugin consists of one or multiple packages in the
+`plugins/` directory. Up till now, we have a simple conventions for naming
+plugin packages: Plugins are named `x`, with the option of having a related
+backend plugin called `x-backend` (where `x` is the plugin name, like `catalog`
+or `techdocs`). There is a need for sharing code between the frontend and
+backend of a plugin, between backend plugins, or components and hooks between
+different frontend plugins
+([some examples](https://github.com/backstage/backstage/issues/3655#issuecomment-758166746)).
+This results in emerging plugin packages with shared code, like
+`packages/catalog-client` or `packages/techdocs-common`.
+
+> There is a common prase in software development:
+> [Naming things is hard](https://martinfowler.com/bliki/TwoHardThings.html)
+
+To keep the contributed plugins consistent, this Architecture Decision Record
+provides rules for naming plugin packages.
+
+## Decision
+
+We will place all plugin related code in the `plugins/` directory. The
+`packages/` directory is reserved for core package of Backstage.
+
+We follow this structure for plugin packages (where `x` is the plugin name, for
+example `catalog` or `techdocs`):
+
+- `x`: Contains the main frontend code of the plugin.
+- `x-backend`: Contains the main backend code of the plugin.
+- `x-common-react`: Contains shared widgets, hooks and similar that both the
+  plugin itself (`x`) and third-party frontend plugins can depend on.
+- `x-common-node`: Contains utilities for backends that both the plugin backend
+  itself (`x-backend`) and third-party backend plugins can depend on.
+- `x-common`: An isomorphic package with platform agnostic models, clients, and
+  utilities that all packages above or any third-party plugin package can depend
+  on.
+
+We prefix the package names with `@backstage/plugin-`.
+
+This structure is based on a suggestion
+[here](https://github.com/backstage/backstage/issues/3655#issuecomment-758166746).
```

Comment:
```suggestion
This structure is based on a [suggestion in issue
#3655](https://github.com/backstage/backstage/issues/3655#issuecomment-758166746).
```

In general, trying to avoid making the word "here" or "click here" clickable is preferred for sight reading (where your eye scans for underscored words or phrases) and accessibility readers (where they're read relevant links, and the word "here" won't make sense in context).